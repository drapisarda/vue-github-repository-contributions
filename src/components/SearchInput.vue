<template>
  <input
    class="form-control"
    type="search"
    :placeholder="placeholder"
    @keyup="$emit('text', $event.target.value)"
  />
</template>

<script>
export default {
  name: 'SearchInput',
  props: {
    placeholder: {
      type: String,
      required: false,
      default: '',
    },
  },
};
</script>
