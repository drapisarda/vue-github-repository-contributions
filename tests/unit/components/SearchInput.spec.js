import { shallowMount } from '@vue/test-utils';
import SearchInput from '@/components/SearchInput.vue';

describe('SearchInput', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(SearchInput);
  });

  test('renders component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders required children', () => {
    const input = wrapper.find('input');

    expect(input.exists()).toBe(true);
  });

  test('renders placeholder props', () => {
    const placeholder = 'It should be a placeholder text';
    const input = wrapper.find('input');

    wrapper.setProps({
      placeholder,
    });

    expect(input.attributes('placeholder')).toBe(placeholder);
  });

  test('emits input text', () => {
    const input = wrapper.find('input');
    const inputValue = 'someuser';

    input.element.value = inputValue;
    input.trigger('keyup');

    expect(wrapper.emitted('text')[0][0]).toBe(inputValue);
  });
});
