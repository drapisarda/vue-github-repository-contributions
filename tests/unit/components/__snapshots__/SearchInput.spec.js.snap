// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders component 1`] = `<input type="search" placeholder="" class="form-control">`;
